charts/backstage/templates/configmap.yaml

```diff
@@ -0,0 +1,23 @@
+{{- range $extraAppConfig := .Values.backstage.extraAppConfig }}
+{{- $configMap := lookup "v1" "ConfigMap" $.Release.Namespace $extraAppConfig.configMapRef  -}}
```

Comment:
I'm not sure we want this behavior. In case the lookup locates a `ConfigMap` with matching name and namespace, it just assumes the configmap is correct, contains proper data. Imagine user updating the `.data` for a particular app config in `values.yaml`. Lookup just checks if the configmap is present, and since it is (despite it contains old data) it just assumes everything is great and upgrades the chart. The chart appears upgraded but the data in configmap is stale.

Why do we want to check for configmap presence here if we want to ensure the data is valid and up to date?

Reply:
This check is to for the existence of  the configMapRef used in the values file, i.e. app-config or app-config.production and use the 'data' only if it's provided in the values file

Reply:
Does that assume the config map was created outside of this chart then?

Let me rephrase my comment above: This lookup breaks chart upgrades. Because if you change data within the configmap (the configmap content is defined in `values.yaml`) and try to upgrade the chart, new data doesn't get propagated.

Imagine following scenario:

1. Have an empty namespace (aka no configmap `bar` exists)
1. Define values like this:
   ```yaml
   # values.yaml
   ...
     extraAppConfig:
     -  filename: foo
        configMapRef: bar
        data: |
          app:
            title: Title
   ...
   ```
2. Now install the chart `helm upgrade -i release_a backstage/backstage`
3. Observe deployed configmap `bar`:
   ```
   apiVersion: v1
   kind: ConfigMap
   metadata:
     name: bar
     ...
   data:
     foo: |
       app:
         title: Title
   ```
4. Change the `values.yaml`:
   ```diff
     # values.yaml
     ...
       extraAppConfig:
       -  filename: foo
          configMapRef: bar
          data: |
            app:
   -          title: Title
   +          title: New Title
     ...
   ```
5. Upgrade the chart `helm upgrade -i release_b backstage/backstage`
6. Observe the configmap, which now contains the stale data (because of the lookup - it found the existing configmap, therefore no configmap was submitted to kube api)
   ```
   apiVersion: v1
   kind: ConfigMap
   metadata:
     name: bar
     ...
   data:
     foo: |
       app:
         title: Title
   ```


If the aim here is to support external configmaps, let's have them fully external and not mash them together with data templated from `values.yaml`. We don't need no lookup here - in that case this is fully user's responsibility to ensure their existence.

If the aim here is to support configmaps defined by data in `values.yaml`, we don't need lookups because the desired state is defined by the chart and it's values, not by the actual state on cluster. Cluster data are not helpful here at all.